src/rgw/rgw_op.cc

```diff
@@ -3095,6 +3105,10 @@ void RGWCreateBucket::execute()
     s->bucket_info.swift_ver_location = *swift_ver_location;
     s->bucket_info.swift_versioning = (! swift_ver_location->empty());
   }
+  if (obj_lock_enabled) {
+    info.flags = BUCKET_VERSIONED | BUCKET_OBJ_LOCK_ENABLED;
+  }
```

Comment:
thinking about the `!is_meta_master()` case of bucket creation, where the master zone is running an older release that doesn't support `x-amz-bucket-object-lock-enabled`. after `forward_request_to_master()`, we decode the master zone's bucket info into `master_info`, so we can check that `master_info.obj_lock_enabled() == obj_lock_enabled` here. if they don't match, how should we handle it? either:
* return an error to the client (even though the master already created the bucket), or
* reset `obj_lock_enabled=false` here to match the result on the master zone

what do you think?

Reply:
I think the second choice maybe better..